estate/models/estate_property.py

```diff
@@ -0,0 +1,110 @@
+from odoo import fields, models, api
+from odoo.exceptions import UserError, ValidationError
+from datetime import date
+from dateutil.relativedelta import relativedelta
+from odoo.tools.float_utils import float_compare
+
+
+class EsateProperty(models.Model):
+    _name = 'estate.property'
+    _description = 'Defines a real estate property'
+
+    name = fields.Char('Title', required=True, index=True)
+    description = fields.Text('Description')
+    postcode = fields.Char('Postcode')
+    date_availability = fields.Date('Available From', copy=False, 
+                                    default=lambda _: date.today() + relativedelta(months=3))
+    expected_price = fields.Float('Expected Price', required=True)
+    selling_price = fields.Float('Selling Price', readonly=True, copy=False, compute='_compute_selling_price', store=True)
+    bedrooms = fields.Integer('Bedrooms', default=2)
+    living_area = fields.Integer('Living Area (sqm)')
+    facades = fields.Integer('Facades')
+    garage = fields.Boolean('Garage')
+    garden = fields.Boolean('Garden')
+    garden_area = fields.Integer('Garden Area (sqm)')
+    garden_orientation = fields.Selection([
+            ('north', 'North'),
+            ('south', 'South'),
+            ('east', 'East'),
+            ('west', 'West')
+        ],
+        'Garden Orientation')
+    active = fields.Boolean('Active', default=True)
+    status = fields.Selection([
+        ('new', 'New'),
+        ('offer_received', 'Offer Received'),
+        ('sold', 'Sold'),
+        ('cancelled', 'Cancelled')
+    ], 'Status', default='new')
+
+    property_type_id = fields.Many2one('estate.property.type', string='Property Type')
+
+    buyer_id = fields.Many2one('res.partner', string='Buyer')
+    salesperson_id = fields.Many2one('res.users', string='Salesperson', default=lambda self: self.env.user)
+
+    property_tag_ids = fields.Many2many('estate.property.tag', string='Property Tags')
+
+    offer_ids = fields.One2many('estate.property.offer', 'property_id', string='Offers')
+
+    total_area = fields.Integer('Total Area (sqm)', compute='_compute_total_area')
+
+    best_offer = fields.Float('Best Offer', compute='_compute_best_offer')
+
+    accepted_offer_id = fields.Many2one('estate.property.offer', string='Accepted Offer')
```

Comment:
This field is not mentioned in the tutorial.

Reply:
answered below